sdks/java/extensions/sql/datacatalog/build.gradle

```diff
@@ -20,13 +20,23 @@ import groovy.json.JsonOutput
 
 plugins { id 'org.apache.beam.module' }
 
-applyJavaNature(automaticModuleName: 'org.apache.beam.sdk.extensions.sql.datacatalog')
+applyJavaNature(
+  enableStrictDependencies: true,
+  automaticModuleName: 'org.apache.beam.sdk.extensions.sql.datacatalog')
 
 dependencies {
   compile enforcedPlatform(library.java.google_cloud_platform_libraries_bom)
   compile(library.java.google_cloud_datacatalog_v1beta1) {
     exclude group: 'io.grpc', module: 'grpc-core' // Use Beam's version
   }
+  compile library.java.gax
+  compile library.java.google_auth_library_credentials
+  compile library.java.protobuf_java
+  compile library.java.slf4j_api
+  compile library.java.vendored_guava_26_0_jre
+  compile project(path: ":sdks:java:core", configuration: "shadow")
+  compile "com.google.api.grpc:proto-google-cloud-datacatalog-v1beta1:0.32.1"
```

Comment:
We should probably used the bom for this, like we discussed on the PR for GCP IO.

Reply:
@ibzib the declaration in BeamModulePlugin is 
`google_cloud_datacatalog_v1beta1            : "com.google.cloud:google-cloud-datacatalog", // google_cloud_platform_libraries_bom sets version`
whereas what's used here is:
`"com.google.api.grpc:proto-google-cloud-datacatalog-v1beta1:0.32.1"`
which doesn't exist in the BOM
https://storage.googleapis.com/java-cloud-bom-dashboard/com.google.cloud/google-cloud-bom/0.147.0/index.html

Reply:
Cancel the above comment. I see there is a layer of dependencies that can also be referred to using the BOM.